javagen/src/main/java/com/azure/autorest/implementation/ClientModelPropertiesManager.java

```diff
@@ -57,8 +57,11 @@ public final class ClientModelPropertiesManager {
     private final LinkedHashMap<String, ClientModelProperty> readOnlyProperties;
     private final ClientModelProperty additionalProperties;
     private final ClientModelPropertyWithMetadata discriminatorProperty;
+    private final boolean allPolymorphicModelsInSamePackage;
```

Comment:
Is this just the result of `PolymorphicDiscriminatorHandler.isAllPolymorphicModelsInSamePackage(model)`? 
It's seems weird to me as field of `ClientModelPropertiesManager`, can we put it in `ClientModel`?

Reply:
I should add some documentation around this. It doesn't work well being in `ClientModel`, as when the `ClientModel` is initialized the parents or children in the structure may not exist yet. That said, I could make this something that is set on access later on.